spring-rabbit/src/main/java/org/springframework/amqp/rabbit/connection/AbstractConnectionFactory.java

```diff
@@ -66,6 +70,10 @@
 
 	private volatile int closeTimeout = DEFAULT_CLOSE_TIMEOUT;
 
+	private ConnectionNameStrategy connectionNameStrategy =
+			connectionFactory -> (this.beanName != null ? this.beanName : getClass()) +
+					"#" + this.defaultConnectionNameStrategyCounter.getAndIncrement();
```

Comment:
I don't think we should use `toString()` here - it adds an unnecessary `class ` prefix:

    class org.springframework.amqp.rabbit.connection.CachingConnectionFactory#0

Also, the FQCN is too much - maybe just simple name? Or `springAMQP#0` ??